pkg/autoscaler/stats_scraper_test.go

```diff
@@ -149,28 +192,61 @@ func TestScrapeViaURL_ErrorCases(t *testing.T) {
 	}
 }
 
-func TestSendStatMessage(t *testing.T) {
-	metric := getTestMetric()
+func TestScrape_HappyCase(t *testing.T) {
 	client := newTestClient(getHTTPResponse(200, testAverageConcurrenyContext+testQPSContext), nil)
-	scraper, err := newServiceScraperWithClient(&metric, TestLogger(t), client)
+	scraper, err := serviceScraperForTest(client)
 	if err != nil {
 		t.Errorf("newServiceScraperWithClient=%v, want no error", err)
 	}
 
-	wantConcurrency := 2.1
-	stat := Stat{AverageConcurrentRequests: wantConcurrency}
+	// Make an Endpoints with 2 pods.
+	createEndpoints(addIps(makeEndpoints(), 2))
+
 	statsCh := make(chan *StatMessage, 1)
 	defer close(statsCh)
-	scraper.sendStatMessage(stat, statsCh)
+	scraper.Scrape(TestContextWithLogger(t), statsCh)
 
 	got := <-statsCh
 	if got.Key != testKPAKey {
 		t.Errorf("StatMessage.Key=%v, want %v", got.Key, testKPAKey)
 	}
-	if got.Stat.AverageConcurrentRequests != wantConcurrency {
+	if got.Stat.AverageConcurrentRequests != 2.0 {
 		t.Errorf("StatMessage.Stat.AverageConcurrentRequests=%v, want %v",
-			got.Stat.AverageConcurrentRequests, wantConcurrency)
+			got.Stat.AverageConcurrentRequests, 2.0)
 	}
+	// 2 pods times 2.0
+	if got.Stat.AverageRevConcurrency != 4.0 {
+		t.Errorf("StatMessage.Stat.AverageRevConcurrency=%v, want %v",
+			got.Stat.AverageRevConcurrency, 4.0)
+	}
+}
+
+func TestScrape_DoNotScrapeIfNoPodsFound(t *testing.T) {
+	client := newTestClient(getHTTPResponse(200, testAverageConcurrenyContext+testQPSContext), nil)
+	scraper, err := serviceScraperForTest(client)
+	if err != nil {
+		t.Errorf("newServiceScraperWithClient=%v, want no error", err)
```

Comment:
this should probably be Fatalf, that is if you got an error here, you wouldnt bother proceeding with the rest of the test.

Reply:
Done.